pymongo/daemon.py

```diff
@@ -64,8 +65,10 @@ def _spawn_daemon(args):
                     stdin=devnull, stderr=devnull, stdout=devnull)
                 _silence_resource_warning(popen)
         except FileNotFoundError as exc:
-            raise PyMongoError(
-                f'Failed to start {args[0]}: is it on your $PATH?\nOriginal exception: {exc}')
+            warnings.warn(f'Failed to start {args[0]}: is it on your $PATH?\n'
+                          f'Original exception: {exc}', RuntimeWarning,
+                          stacklevel=2)
+            return subprocess.CompletedProcess(args, 0)
```

Comment:
What's the point of `return subprocess.CompletedProcess(args, 0)`? The success case doesn't return anything.

Reply:
See my comments above... The success case returns the output of `subprocess.Popen`.

Reply:
Another option is to check that `_spawn` returned non-`None` before trying to set the `returncode` in `suppress_resource_warning`

Reply:
Look again, `_spawn_daemon` (Windows) doesn't return anything on success, but `_spawn` (unix) does. It's odd to return `CompletedProcess` in the first case.

Reply:
Ah got it. I changed it so that `_silence_resource_warning` only sets returncode when it gets a non-None `Popen`.